src/TogglePassword/src/Form/TogglePasswordTypeExtension.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TogglePassword\Form;
+
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Félix Eymonot <felix.eymonot@alximy.io>
+ */
+final class TogglePasswordTypeExtension extends AbstractTypeExtension
+{
+    public static function getExtendedTypes(): iterable
+    {
+        return [PasswordType::class];
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefined([
+            'toggle',
+            'hidden_label',
+            'visible_label',
+            'hidden_icon',
+            'visible_icon',
+        ]);
+        $resolver->setAllowedTypes('toggle', ['boolean']);
+        $resolver->setAllowedTypes('hidden_label', ['string', 'null']);
+        $resolver->setAllowedTypes('visible_label', ['string', 'null']);
+        $resolver->setAllowedTypes('hidden_icon', ['string', 'null']);
+        $resolver->setAllowedTypes('visible_icon', ['string', 'null']);
+        $resolver->setDefaults([
+            'toggle' => false,
+            'hidden_label' => 'Hide',
+            'visible_label' => 'Show',
+            'hidden_icon' => 'Default',
+            'visible_icon' => 'Default',
+        ]);
+    }
+
+    public function finishView(FormView $view, FormInterface $form, array $options): void
+    {
+        if (!$options['toggle']) {
+            return;
+        }
+
+        $attr = $view->vars['attr'] ?? [];
+        $controllerName = 'symfony--ux-toggle-password--toggle-password';
+        $attr['data-controller'] = trim(sprintf('%s %s', $attr['data-controller'] ?? '', $controllerName));
+
+        $controllerValues = [];
+        $controllerValues['hidden-label'] = $options['hidden_label'] ?? null;
+        $controllerValues['visible-label'] = $options['visible_label'] ?? null;
+        $controllerValues['hidden-icon'] = $options['hidden_icon'] ?? null;
+        $controllerValues['visible-icon'] = $options['visible_icon'] ?? null;
```

Comment:
There are default values, so `?? null` is not required I guess.

Reply:
That's right, but default values are not `null`, and if you choose to set `null` for one of these options (e.g to only display the icon) then the option is unset of the `$options` array. So i used `?? null `to ensure that `null` values are passed to the controller values

Reply:
I haven't tested in the context of a Form extension, but in a Form type, passing `null` as a value does not unset the option (just tested right now).

If the option is really unset, that I think it should be considered a bug in Symfony? I mean, `null` is still a real value.

Reply:
Must have missed it but after checking again, you're right ! 
With default values set, if we pass `null` from our Form, the option isn't unset we get the `null` value 👍🏼